### PR TITLE
Fix OpsGenie notifier and add unit tests

### PR DESCRIPTION
**Pull request by iksaif**
See #1223, looks like OpsGenie now sometimes returns a 422 when you
don't specify a team. This change cleans up the JSON output and
add a few unit tests.

### CODE REVIEW

File: notify/impl.go [hunk 1]
```diff
@@ -987,45 +1013,39 @@ func (n *OpsGenie) Notify(ctx context.Context, as ...*types.Alert) (bool, error)
 
 		apiURL = n.conf.APIURL + "v2/alerts"
 		var teams []map[string]string
-		for _, t := range strings.Split(string(tmpl(n.conf.Teams)), ",") {
+		for _, t := range safeSplit(string(tmpl(n.conf.Teams)), ",") {
```
Comment (stuartnelson3) [inline notify/impl.go]: I think this should be returned to just `strings.Split` and then check each `t` for being an empty string. If someone enters `"team1,"` in their config, it'll still end up adding `"name":""` in the json.

Comment (iksaif) [reply notify/impl.go]: done

File: notify/impl.go [hunk 2]
```diff
@@ -952,9 +952,39 @@ type opsGenieCloseMessage struct {
...
+		return true, err
+	}
+	defer resp.Body.Close()
+
+	return n.retry(resp.StatusCode)
+}
+
+// Like Split but filter out empty strings.
+func safeSplit(s string, sep string) []string {
+	// https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating
```
Comment (stuartnelson3) [inline notify/impl.go]: this can be removed

Comment (iksaif) [reply notify/impl.go]: done

**Comment by dswarbrick** [comment]
Is there some reason why Alertmanager doesn't simply use https://github.com/opsgenie/opsgenie-go-sdk?

**Comment by stuartnelson3** [comment]
Not that I'm aware of. Potentially it didn't exist when this was written.

Also, the entire implementation in our codebase is ~150 lines of code. Maybe it wasn't viewed as necessary to add the dependency.

**Comment by brian-brazil** [comment]
I don't believe we were aware of it. We had problems even finding what API docs there were.

**Pull request merged**